Core/ExposureNPC.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.ModLoader;
+
+namespace StarlightRiver.Core
+{
+    public class ExposureNPC : GlobalNPC
+    {
+        public int ExposureAdd;
+        public int ExposureAddMelee;
+        public int ExposureAddRanged;
+        public int ExposureAddMagic;
+        public int ExposureAddSummon; //literally just summon tag without the requirement of being the players minion target
+
+        public float ExposureMult;
+        public float ExposureMultMelee;
+        public float ExposureMultRanged;
+        public float ExposureMultMagic;
+        public float ExposureMultSummon;
+
+        public override bool InstancePerEntity => true;
+        public override void ResetEffects(NPC npc)
+        {
+            ExposureAdd = 0;
```

Comment:
Maybe this should be named ``ExposureAddAll`` or something to explicitly clarify its global.